app/models/photo.rb

```diff
@@ -81,12 +81,12 @@ def ownership_of_status_message
     end
   end
 
-  def self.diaspora_initialize(params = {})
-    photo = self.new params.to_hash.slice(:text, :pending)
-    photo.author = params[:author]
-    photo.public = params[:public] if params[:public]
-    photo.pending = params[:pending] if params[:pending]
+  class << self
+    alias_method :diaspora_initialize_shareable, :diaspora_initialize
+  end
```

Comment:
I don't like this, and I don't know if the DRY is worth this? You do this only to not write this twice?
```ruby
new(params.to_hash.stringify_keys.slice(*column_names)).tap do |photo|
  photo.author = params[:author]
```